src/Tesseract/FontInfo.cs

```diff
@@ -81,20 +43,23 @@ private bool HasFlag<T>(T value, T flag)
             return null;
         }
 
+        public static object _cacheLock = new object();
         public static FontInfo GetOrCreate(
             string name, int id,
             bool isItalic, bool isBold, bool isFixedPitch,
             bool isSerif, bool isFraktur = false
         )
         {
-            if (_cache.ContainsKey(id)) {
-                return _cache[id];
-            }
+            lock (_cacheLock) {
```

Comment:
Your correct in that this solves the threading safety issue however using a static dictionary still has some other issues in this case the main one would be it's possible, perhaps even likely, that the Id's may not be unique across processing runs or engines. For instance if you process two seperate documents it's possible that the same id could be returned with different font information. Looking at the doco for LTRResultIterator::WordFontAttributes confirms this "Lifespan is the same as the iterator itself, ie rendered invalid by various members of TessBaseAPI, including Init, SetImage, End or deleting the TessBaseAPI."

I think the best solution would be to move it as a cache in ResultIterator as previously mentioned, this also wouldn't require any thread synchronisation\locking since ResultIterator can only be used by one thread.